ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoSignedTransaction.kt

```diff
@@ -0,0 +1,54 @@
+package net.corda.v5.ledger.utxo.transaction
+
+import net.corda.v5.application.crypto.DigitalSignatureAndMetadata
+import net.corda.v5.application.serialization.SerializationService
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.SecureHash
+import java.security.PublicKey
+
+/**
+ * Defines a signed UTXO transaction.
+ *
+ * @property id The ID of the transaction.
+ * @property signatures The signatures that have been applied to the transaction.
+ */
+@DoNotImplement
+@CordaSerializable
+interface UtxoSignedTransaction {
+
+    val id: SecureHash
+    val signatures: Set<DigitalSignatureAndMetadata>
+
+    /**
+     * Adds the specified signatures to the current [UtxoSignedTransaction].
+     *
+     * @param signatures The signatures and metadata to add to the current [UtxoSignedTransaction].
+     * @return Returns a new [UtxoSignedTransaction] containing the applied signatures.
+     */
+    fun addSignatures(signatures: Iterable<DigitalSignatureAndMetadata>): UtxoSignedTransaction
+
+    /**
+     * Adds the specified signature to the current [UtxoSignedTransaction].
+     *
+     * @param signatures The signatures and metadata to add to the current [UtxoSignedTransaction].
+     * @return Returns a new [UtxoSignedTransaction] containing the applied signature.
+     */
+    fun addSignatures(vararg signatures: DigitalSignatureAndMetadata): UtxoSignedTransaction
+
+    /**
+     * Gets the missing signatories from the current [UtxoSignedTransaction].
+     *
+     * @param serializer The [SerializationService] required to obtain missing signed transaction signatures.
+     * @return Returns a [Set] of [PublicKey] representing the missing signatories from the current [UtxoSignedTransaction].
+     */
+    fun getMissingSignatories(serializer: SerializationService): Set<PublicKey>
```

Comment:
Interesting question if serializer should be passed in. It is done to allow user specified serializers, but question is if we are considering that in our logic. I think at this point it is probably too late to change your mind as TxId needs to be same as one in builder. I do think you might need this though if you did need custom formats. Need for this has also been reduced recently, so agin perhaps we defer this and just provide the 'defaulted' no parameter form